src/components/drive-container.tsx

```diff
@@ -0,0 +1,136 @@
+import {
+    ActionType,
+    DriveTreeItem,
+    DriveView,
+    ItemType,
+    OnItemOptionsClickHandler,
+    TreeItem,
+    traverseTree,
+} from '@powerhousedao/design-system';
+import { Drive, Node } from 'document-model-libs/document-drive';
+import { Immutable } from 'document-model/document';
+import { useEffect, useState } from 'react';
+import { useDocumentDrive } from 'src/hooks/useDocumentDrive';
+
+function mapDocumentDriveNodeToTreeItem(
+    node: Immutable<Node>,
+    allNodes: Immutable<Array<Node>>
+): TreeItem {
+    const isFolder = node.kind === 'folder';
+    const pathRegex = new RegExp(`^${node.path}/[^/]+`);
+
+    return {
+        id: node.path,
+        label: node.name,
+        type: isFolder ? ItemType.Folder : ItemType.File,
+        children: isFolder
+            ? allNodes
+                  .filter(childrenNode => pathRegex.test(childrenNode.path))
+                  .map(childrenNode =>
+                      mapDocumentDriveNodeToTreeItem(childrenNode, allNodes)
+                  )
+            : undefined,
+    };
+}
+
+function mapDocumentDriveToTreeItem(drive: Immutable<Drive>): DriveTreeItem {
+    return {
+        id: drive.id,
+        label: drive.name,
+        type: ItemType.LocalDrive,
+        expanded: true,
+        children: drive.nodes
+            .filter(node => !node.path.includes('/'))
+            .map(node => mapDocumentDriveNodeToTreeItem(node, drive.nodes)),
+    };
+}
+
+export default function () {
+    const [drives, setDrives] = useState<DriveTreeItem[]>([]);
+    const { documentDrive, openFile, addFolder, deleteNode, renameNode } =
+        useDocumentDrive();
+
+    useEffect(() => {
+        const newItems = documentDrive?.state.drives.map(
+            mapDocumentDriveToTreeItem
+        );
+        setDrives(newItems ?? []);
+    }, [documentDrive]);
+
+    function handleNodeClick(item: TreeItem, drive: DriveTreeItem) {
+        if (item.type === ItemType.File) {
+            openFile(drive.id, item.id);
+        }
+    }
+
+    function addNewFolder(item: TreeItem, drive: DriveTreeItem) {
+        const basePath = item.id.replace(drive.id, '');
+        addFolder(
+            drive.id,
+            `${basePath ? `${basePath}/` : ''}new-folder`, // TODO check if there is a new folder already
```

Comment:
you can use `path.join` here instead

Reply:
I've added this implementation in this [PR](https://github.com/powerhouse-inc/document-model-electron/pull/10)

Reply:
actually you can use native modules in the browser. I think you have to disable manually, but rn is enabled

Reply:
But there is still the web bundle which wont run inside electron.
Created this issue #11 so we can take care of this later.

Reply:
that's right, maybe we can keep the `path` polyfill, it's useful